singularity-eos/eos/eos_gruneisen.cpp

```diff
@@ -49,45 +52,48 @@ PORTABLE_FUNCTION Real Gruneisen::ComputeRhoMax(const Real s1, const Real s2, co
   // Polynomial from the denominator of the reference pressure curve:
   auto poly = [=](Real eta) { return 1 - s1 * eta - s2 * square(eta) - s3 * cube(eta); };
 
+  // Comparisons to zero are actually made to
+  static constexpr Real EPS = std::numeric_limits<Real>::epsilon();
+
   // First find the eta root. A negative root indicates that there is no maximum density.
-  Real root = 0; // Non-sensical root means a root hasn't been found yet
+  Real root = -1.; // Non-sensical root means none exists
   Real discriminant, root1, root2;
-  if (s2 == 0 && s3 == 0 && s1 > 0) {
+  if (is_near_zero(s2, EPS) && is_near_zero(s3, EPS) && s1 > EPS) {
```

Comment:
are the s's all guaranteed to be positive? i.e., should this be `std::abs(s1) > EPS`?

Reply:
I would first argue that there is no guarantee since we don't enforce anything about the selection of the s values so I wouldn't assume anything about sign.

But in this case the critical test is that `s1` is positive, not that it is nonzero. If `s1` is negative, then the root would be negative anyway and it's not relevant.

However, you do remind me that if `s1 < 1`, the root would be greater than 1 anyway which is nonsensical. I'm not guarding against that though so really the condition should be `s1 > 1.`. I'll fix that

Reply:
sounds good.